big-o-notation/big-o.js

```diff
@@ -0,0 +1,44 @@
+/* exported uniqueQuadratic, uniqueLinear */
+
+function uniqueLinear(words) {
+  const seen = {};                  // 1 * 1 = O(1)
+  const unique = [];                // 1 * 1 = O(1)
+  for (
+    let i = 0;                      // 1 * 1 = O(1)
+    i < words.length;               // 2 * n = O(2n) ~= 0(n)
+    i++                             // 2 * n = O(2n) ~= 0(n)
+  ) {
+    const word = words[i];          // 3 * n = O(3n) ~= 0(n)
+    if (!seen[word]) {              // 2 * n = O(2n) ~= 0(n)
+      seen[word] = true;
+      unique[unique.length] = word; // 3 * n = O(3n) ~= 0(n)
+    }
+  }
+  return unique;
+} // Big O Notation for uniqueLinear: O(n)
+
+function uniqueQuadratic(words) {
+  const unique = [];                // 1 * 1 = O(1)
+  for (
+    let i = 0;                      // 1 * 1 = O(1)
+    i < words.length;               // 2 * n = O(2n) ~= 0(n)
+    i++                             // 2 * n = O(2n) ~= 0(n)
+  ) {
+    const word = words[i];          // 3 * n = O(3n) ~= 0(n)
+    let isUnique = true;            // 1 * 1 = O(1)
+    for (
+      let c = 0;                    // 1 * 1 = O(1)
+      c < i;                        // 2 * n = O(2n) ~= 0(n)
+      c++                           // 2 * n = O(2n) ~= 0(n)
+    ) {
+      const comparing = words[c];   // 3 * n = O(3n) ~= 0(n)
+      if (comparing === word) {     // 2 * n = O(2n) ~= 0(n)
+        isUnique = false;           // 1 * 1 = O(1)
+      }
```

Comment:
Worst-case scenario these statements are executed once for each word, for each word. So since this inner loop potentially loops through every word again, we're looking at O(n^2)